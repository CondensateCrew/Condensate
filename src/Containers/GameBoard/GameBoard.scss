@import '../../css/variables.scss';

@function random_range($min, $max) {
  $rand: random();
  $random_range: $min + floor($rand * (($max - $min) + 1));
  @return $random_range;
}

.game-board {
  &>section {
    height: 100%;
    position: relative;
    &.limited-bbls {
      display: -webkit-flex;
      display: -ms-flex;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: space-between;
      &>section {
        display: -webkit-flex;
        display: -ms-flex;
        display: flex;
        align-items: center;
        justify-content: center;
        height: 100%;
      }
      .bubble {
        margin: 0 2*$large-space;
        width: 200px; height: 200px;
      }
    }
  }
  .bubble {
    animation: appearing 8s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;
    box-sizing: border-box;
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    align-items: center;
    justify-content: center;
    background: $primary-gradient;
    border-radius: 50%;
    box-shadow: $doubled-shadow;
    font-size: $big-text;
    line-height: 1em;
    width: fit-content;
    z-index: 20;
    button {
      cursor: pointer;
      padding: 0 $small-space;
    }
    &.randomly {
      position: absolute;
      $total: 8;
      @for $i from 1 through $total {
        &:nth-child(#{$i}) {
          $new: $i - 1;
          animation-delay: $i * 0.5s;
          opacity: 0;
        }
      }
      &:nth-child(1) {
        top: 5%; left: 5%;
      }
      &:nth-child(4) {
        top: 12%; left: 35%;
      }
      &:nth-child(5) {
        top: 38%; left: 87%;
      }
      &:nth-child(6) {
        top: 5%; left: 65%;
      }
      &:nth-child(2) {
        top: 80%; left: 70%;
      }
      &:nth-child(3) {
        top: 80%; left: 25%;
      }
      &:nth-child(7) {
        top: 53%; left: 48%;
      }
      &:nth-child(8) {
        top: 50%; left: 8%;
      }
      &:after {
        content: "";
        display: block;
        padding-bottom: 100%;
        width: 100%;
      }
    }
  }
}

.bubble-rounds {
  @extend %rounds;
  display: -webkit-flex;
  display: -ms-flex;
  display: flex;
  align-items: center;
  animation-delay: 1.5s;
  justify-content: center;
  width: 75%; height: 75%;
  z-index: -15;
  &::before {
    @extend %rounds;
    animation-duration: 5s;
    animation-delay: 1s;
    width: 40%; height: 40%;
    z-index: -10;
  }
  &::after {
    @extend %rounds;
    animation-delay: 0s;
    width: 10%; height: 10%;
    z-index: -5;
  }
}

.disappeared {
  position: relative;
  background: none !important;
  box-shadow: none !important;
  &::before {
    @extend %rounds;
    animation-delay: 2s;
    width: 100%; height: 100%;
    z-index: -20;
  }
}

@keyframes appearing {
  0% {
    opacity: 0;
    transform: scale(0);
  }
  10% {
    opacity: 1;
    transform: scale(1);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
}

@keyframes disappearing {
  0% {
    opacity: 1;
    transform: scale(1);
  }
  10% {
    opacity: 0;
    transform: scale(0);
  }
  100% {
    opacity: 0;
    transform: scale(0);
  }
}

@keyframes spread {
  0% {
    opacity: 0;
    transform: scale(1);
  }
  30% {
    opacity: 0.6;
  }
  100% {
    opacity: 0;
    transform: scale(2);
  }
}

%rounds {
  content: '';
  border-radius: 50%;
  box-shadow: $doubled-small-shadow;
  position: absolute;
  animation: spread 3s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;
}
