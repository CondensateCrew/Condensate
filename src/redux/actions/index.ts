import {
  Action,
  Category,
  Brainstorm,
  UserLoginReceived,
  RandomWordCollection,
  Insight,
  ActionObject,
  IAddUserAction,
  ILogoutUserAction,
  IAddAllActionsAction,
  IAddNewActionAction,
  IAddAllBrainstormsAction,
  IAddNewBrainstormAction,
  IDeleteBrainstormAction,
  IAddAllCategoriesAction,
  IAddNewCategoryAction,
  IAddRandomWordCollectionsAction,
  IAddQuestionTemplatesAction,
  IAddSecretSauceAction,
  IAddCurrentBrainstormAction,
  IRemoveCurrentBrainstormAction,
  IAddChosenWordAction,
  IRemoveChosenWordAction,
  IAddInsightAction,
  IAddQueryAction,
  IRemoveQueryAction,
  IAddFilterAction,
  IRemoveFilterAction,
  IEndTimeAction,
  IReverseTimeAction,
  IEndInstructionsAction,
  IReverseInstructionsAction
} from 'interfaces';

// user reducer action creators
export const addUser = (user: UserLoginReceived): IAddUserAction => ({
  type: 'ADD_USER',
  user: user
});

export const logOutUser = (): ILogoutUserAction => ({
  type: 'LOG_OUT_USER',
});

// actions reducer action creators
export const addAllActions = (actions: Action[]): IAddAllActionsAction => ({
  type: 'ADD_ALL_ACTIONS',
  actions: actions
});

export const addNewAction = (action: Action): IAddNewActionAction => ({
  type: 'ADD_NEW_ACTION',
  action: action
});

// allBrainstorms reducer action creators
export const addAllBrainstorms = (brainstorms: Brainstorm[]): IAddAllBrainstormsAction => ({
  type: 'ADD_ALL_BRAINSTORMS',
  brainstorms: brainstorms
});

export const addNewBrainstorm = (brainstorm: Brainstorm): IAddNewBrainstormAction => ({
  type: 'ADD_BRAINSTORM',
  brainstorm: brainstorm
});

export const deleteBrainstorm = (id: number): IDeleteBrainstormAction => ({
  type: 'DELETE_BRAINSTORM',
  id: id
});

// categories reducer action creators
export const addAllCategories = (categories: Category[]): IAddAllCategoriesAction => ({
  type: 'ADD_ALL_CATEGORIES',
  categories: categories
});

export const addNewCategory = (category: Category): IAddNewCategoryAction => ({
  type: 'ADD_CATEGORY',
  category: category
});

// randomWordCollections reducer action creators
export const addRandomWordCollections = (collections: RandomWordCollection[]): IAddRandomWordCollectionsAction => ({
  type: 'ADD_ALL_COLLECTIONS',
  collections: collections
});

// questionTemplates reducer action creators
export const addQuestionTemplates = (templates: string[]): IAddQuestionTemplatesAction => ({
  type: 'ADD_ALL_TEMPLATES',
  templates: templates
});

// secretSauce reducer action creators
export const addSecretSauce = (secretSauce: string[]): IAddSecretSauceAction => ({
  type: 'ADD_SECRETE_SAUSE',
  secretSauce: secretSauce
});

// currentBrainstorm reducer action creators
export const addCurrentBrainstorm = (currentBrainstorm: Brainstorm): IAddCurrentBrainstormAction => ({
  type: 'ADD_CURRENT_BRAINSTORM',
  currentBrainstorm: currentBrainstorm
});

export const removeCurrentBrainstorm = (): IRemoveCurrentBrainstormAction => ({
  type: 'REMOVE_CURRENT_BRAINSTORM'
});

// chosenWords reducer action creators
export const addChosenWord = (chosenWord: string): IAddChosenWordAction => ({
  type: 'ADD_WORD',
  chosenWord: chosenWord
});

export const removeChosenWord = (chosenWord: string): IRemoveChosenWordAction => ({
  type: 'REMOVE_WORD',
  chosenWord: chosenWord
});

// insights reducer action creators
export const addInsight = (insight: Insight): IAddInsightAction => ({
  type: 'ADD_INSIGHT',
  insight: insight
});

// query reducer action creators
export const addQuery = (query: string): IAddQueryAction => ({
  type: 'ADD_QUERY',
  query: query
});

export const removeQuery = (): IRemoveQueryAction => ({
  type: 'REMOVE_QUERY'
});

// filter reducer action creators
export const addFilter = (filter: string): IAddFilterAction => ({
  type: 'ADD_FILTER',
  filter: filter
});

export const removeFilter = (): IRemoveFilterAction => ({
  type: 'REMOVE_FILTER'
});

// timeEnded reducer action creators
export const endTime = (): IEndTimeAction => ({
  type: 'END_TIME'
});

export const reverseTime = (): IReverseTimeAction => ({
  type: 'REVERSE_TIME'
});

// instructionsEnded reducer action creators
export const endInstructions = (): IEndInstructionsAction => ({
  type: 'END_INSTRUCTIONS'
});

export const reverseInstructions = (): IReverseInstructionsAction => ({
  type: 'REVERSE_INSTRUCTIONS'
});
