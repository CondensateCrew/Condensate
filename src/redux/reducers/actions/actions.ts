import { ActionObject } from 'interfaces';

type state = string[] | [ ];

const actions = (state: state = [ ], action: ActionObject) => {
  switch(action.type) {
    case 'ADD_ALL_ACTIONS':
      return action.actions;
    case 'ADD_NEW_ACTION':
      return [...state, action.action];
    case 'CLEAN_STORE':
      return [ ];
    default:
      return state;
  }
}

export default actions;
