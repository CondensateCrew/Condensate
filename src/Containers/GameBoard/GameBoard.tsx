import React, {useState, useEffect} from 'react';
import { useSelector, useDispatch } from 'react-redux';
import './GameBoard.scss';
import { AppStore } from 'interfaces';
import { endTime } from 'redux/actions';
import BubblesAll from '../BubblesAll/BubblesAll';
import BubblesReducer from '../BubblesReducer/BubblesReducer';

const GameBoard: React.FC = () => {
  const dispatch = useDispatch();
  const chosenWords = useSelector((store: AppStore) => store.chosenWords);

  useEffect(() => {
    if( chosenWords.length === 9) {
      dispatch(endTime());
    }
  }, [ chosenWords ]);

  return (
    <main className="game-board">
      { !(chosenWords.length === 9)
        ? <BubblesAll />
        : <BubblesReducer chosenWords={chosenWords} />
      }
    </main>
  )
}

export default GameBoard;
