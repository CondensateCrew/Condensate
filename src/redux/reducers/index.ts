import { combineReducers } from 'redux';
import user from '../reducers/user/user';
import allBrainstorms from '../reducers/allBrainstorms/allBrainstorms';
import categories from '../reducers/categories/categories';
import actions from '../reducers/actions/actions';
import randomWordCollections from '../reducers/randomWordCollections/randomWordCollections';
import questionTemplates from '../reducers/questionTemplates/questionTemplates';
import secretSauce from '../reducers/secretSauce/secretSauce';
import currentBrainstorm from '../reducers/currentBrainstorm/currentBrainstorm';
import chosenWords from '../reducers/chosenWords/chosenWords';
import insights from '../reducers/insights/insights';
import query from '../reducers/query/query';
import filter from '../reducers/filter/filter';
import timeEnded from '../reducers/timeEnded/timeEnded';
import instructionsEnded from '../reducers/instructionsEnded/instructionsEnded';

const rootReducer = combineReducers({
  user,
  allBrainstorms,
  categories,
  actions,
  randomWordCollections,
  questionTemplates,
  secretSauce,
  currentBrainstorm,
  chosenWords,
  insights,
  query,
  filter,
  timeEnded,
  instructionsEnded
});

export default rootReducer;
