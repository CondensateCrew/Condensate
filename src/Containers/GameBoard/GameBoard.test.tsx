import React from 'react';
import GameBoard from './GameBoard';
import { shallow } from 'enzyme';
import { addChosenWord } from 'redux/actions';
const mockDispatch = jest.fn();

jest.mock('redux/actions');
jest.mock('react-redux', () => ({
  useSelector: () => ({
    chosenWords: ['Rest'],
    timeEnded: false,
  }),
  useDispatch: () => mockDispatch
}));

describe('GameBoard component', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<GameBoard />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call addChosenWord when bubble is clicked', () => {
    wrapper.find('#water').simulate('click', {
      target: {
        parentNode: {
          classList: { add: jest.fn() }
        },
        previousSibling: {
          classList: { add: jest.fn() }
        },
        style: {
          animation: ''
        },
        id: 'water'
      }
    });

    expect(addChosenWord).toHaveBeenCalledWith('water');
  });
});
