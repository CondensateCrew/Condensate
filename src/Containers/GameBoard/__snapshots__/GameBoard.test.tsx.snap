// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoard component should match the snapshot 1`] = `
<main
  className="game-board"
>
  <BubblesAll />
</main>
`;

exports[`GameBoard component should match the snapshot if there is all 9 words 1`] = `
<main
  className="game-board"
>
  <BubblesReducer
    chosenWords={
      Array [
        "Rest",
        "Toy",
        "Snow",
        "Owl",
        "Oil",
        "Grid",
        "Math",
        "Leg",
        "Men",
      ]
    }
  />
</main>
`;
